src/ConfigEditor.test.tsx

```diff
@@ -16,169 +16,185 @@ const setUpMockBackendServer = (mockBackendSrv: { put: () => void; post: () => v
+          post: jest.fn().mockResolvedValue({ externalId: 'fake-external-id' }),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('fake-external-id')).toBeInTheDocument();
+      });
+    
+      it('gracefully handles when the fetch for external id throws an error', async () => {
+        setUpMockBackendServer({
+          put: jest.fn().mockResolvedValue({ datasource: {} }),
+          post: jest.fn().mockRejectedValue('the server exploded for some reason'),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('External Id is currently unavailable')).toBeInTheDocument();
+      });
+    
+      it('gracefully handles when the fetch for external id return an empty string', async () => {
+        setUpMockBackendServer({
+          put: jest.fn().mockResolvedValue({ datasource: {} }),
+          post: jest.fn().mockResolvedValue({ externalId: '' }),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('External Id is currently unavailable')).toBeInTheDocument();
+      });
+    }
+    describe('ConfigEditor with awsDatasourcesNewFormStyling feature toggle disabled', () => {
+      beforeAll(() => {
+        // @ts-ignore
```

Comment:
I should have provided more context here about this, the reason Im setting the tests up like this is because of how jest executes them: https://github.com/grafana/redshift-datasource/pull/255#discussion_r1373445647
But you have a point about the cleanup, will do that
Also, the reason I added a ts-ignore is that sometime in grafana runtime 9 or 10 they removed support for any feature toggles that aren't explicitly named. Aaaand i just realized 10.2.0 has been released this week, so I'll update the dependencies